File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/MusicGenerationApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class MusicGenerationApi(private val client: HttpClient) {

    /** Create music generation task */
    suspend fun create(body: GenerateMusicForm): PlusApiResultGenerateMusicVO? {
        return client.post(ApiPaths.backendPath("/generation/music/create"), body) as? PlusApiResultGenerateMusicVO
    }

    /** Get music generation result */
    suspend fun getResult(taskId: String): PlusApiResultGenerateMusicVO? {
        return client.get(ApiPaths.backendPath("/generation/music/result/$taskId")) as? PlusApiResultGenerateMusicVO
    }
}

File: sdkwork-backend-sdk-csharp/Api/MusicGenerationApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class MusicGenerationApi
    {
        private readonly HttpClient _client;

        public MusicGenerationApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Create music generation task
        /// </summary>
        public async Task<PlusApiResultGenerateMusicVO?> CreateAsync(GenerateMusicForm body)
        {
            return await _client.PostAsync<PlusApiResultGenerateMusicVO>(ApiPaths.BackendPath("/generation/music/create"), body);
        }

        /// <summary>
        /// Get music generation result
        /// </summary>
        public async Task<PlusApiResultGenerateMusicVO?> GetResultAsync(string taskId)
        {
            return await _client.GetAsync<PlusApiResultGenerateMusicVO>(ApiPaths.BackendPath($"/generation/music/result/{taskId}"));
        }
    }
}
